numpy/_core/src/multiarray/methods.c

```diff
@@ -74,36 +74,28 @@ npy_forward_method(
         PyObject *callable, PyObject *self,
         PyObject *const *args, Py_ssize_t len_args, PyObject *kwnames)
 {
-    PyObject *args_buffer[NPY_MAXARGS];
-    /* Practically guaranteed NPY_MAXARGS is enough. */
-    PyObject **new_args = args_buffer;
-
     /*
      * `PY_VECTORCALL_ARGUMENTS_OFFSET` seems never set, probably `args[-1]`
      * is always `self` but do not rely on it unless Python documents that.
      */
     npy_intp len_kwargs = kwnames != NULL ? PyTuple_GET_SIZE(kwnames) : 0;
-    size_t original_arg_size = (len_args + len_kwargs) * sizeof(PyObject *);
-
-    if (NPY_UNLIKELY(len_args + len_kwargs > NPY_MAXARGS)) {
-        new_args = (PyObject **)PyMem_MALLOC(original_arg_size + sizeof(PyObject *));
-        if (new_args == NULL) {
-            /*
-             * If this fails Python uses `PY_VECTORCALL_ARGUMENTS_OFFSET` and
-             * we should probably add a fast-path for that (hopefully almost)
-             * always taken.
-             */
-            return PyErr_NoMemory();
-        }
+    npy_intp total_nargs = (len_args + len_kwargs);
+
+    NPY_ALLOC_WORKSPACE(new_args, PyObject *, 14, total_nargs + 1);
+    if (new_args == NULL) {
+        /*
```

Comment:
This comment seems weird in this place (was weird before too); is it `len_args` that would be the problem, because its top-most bit would be set? If so, why not just take that into account as per python recommendations?

Obviously, can also just ignore this for now, since it definitely was not clear before...

Reply:
Yeah, I think Python never passes that top bit flag to module functions (rather than `tp_vectorcall`).  So when I wrote it first, I wasn't sure I preferred to add it defensively, but was also hedging a bit with the comment...

Tried to make that slightly clearer, but dunno.  Anyway, tempted to ignore it and may merge soon or so if nobody has complains.

(if anyone reads it after merging, happy to follow up and change things)